components/automate-cli/pkg/verifyserver/services/systemuserservice/systemuserservice.go

```diff
@@ -0,0 +1,211 @@
+package systemuserservice
+
+import (
+	"strings"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/fiberutils"
+	"github.com/chef/automate/lib/logger"
+)
+
+type SystemUser interface {
+	GetSystemUserServiceDetails() *models.SystemUserResponse
+}
+
+type SystemUserService struct {
+	exec fiberutils.ExecCmdService
+	user fiberutils.UserCmdService
+	Log  logger.Logger
+}
+
+func NewSystemUserService(log logger.Logger) *SystemUserService {
+	return &SystemUserService{
+		exec: fiberutils.NewExecCmdServiceImp(),
+		user: fiberutils.NewUserCmdServiceImp(),
+		Log:  log,
+	}
+}
+
+func (su *SystemUserService) GetSystemUserServiceDetails() *models.SystemUserResponse {
+	serviceResponse := &models.SystemUserResponse{}
+	serviceResponse.Passed = true
+	serviceResponseArray := []*models.Checks{}
+
+	habUserResponse, isHabUserCreated := su.ValidateHabUser()
+	if !habUserResponse.Passed {
+		serviceResponse.Passed = false
+	}
+	serviceResponseArray = append(serviceResponseArray, habUserResponse)
+
+	habGroupResponse := su.ValidateHabGroup()
+	if !habGroupResponse.Passed {
+		serviceResponse.Passed = false
+	}
+
+	habUserAndGroupMapResponse := su.ValidateHabUserAndGroupMapping()
+	if !habUserAndGroupMapResponse.Passed {
+		serviceResponse.Passed = false
+	}
+
+	if isHabUserCreated {
+		defer su.deleteUser(constants.USERNAME)
+	}
+
+	serviceResponseArray = append(serviceResponseArray, habGroupResponse, habUserAndGroupMapResponse)
+	checks := make([]models.Checks, len(serviceResponseArray))
+	for i, serviceResp := range serviceResponseArray {
+		checks[i] = *serviceResp
+	}
+	serviceResponse.Checks = checks
```

Comment:
here you are converting `[]*models.Checks` to `[]models.Checks`. This can be avoided if you have `[]*models.Checks` directly in the  `SystemUserResponse` struct

Reply:
Done